From b311f25a791637665918e6fdb17e66c7e5005d53 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Tue, 19 Oct 2021 11:28:29 -0700
Subject: [PATCH 142/196] Bump Lotus version to 1.0.3

Includes changes from Bitcoin ABC versions 0.23.8, v0.23.9, v0.24.0
0.24.1, 0.24.2:

v0.23.8
 - Add a new option `-networkactive` to enable all P2P network activity
   (default 1). To start a node offline, you can provide
   `-networkactive=0` or `-nonetworkactive`.
 - The deprecated `setexcessiveblock` RPC has been removed and is no
   longer available. The `-excessiveblocksize` option should be used
   instead.

v0.23.9:
  - Improvements to the Avalanche experimental feature.
  - Test performance improvements.

v0.24.0
 - The `-ecash` setting is now **enabled by default**. To use the old
   BCHA default unit and bitcoincash: CashAddr prefix, use the
   `-ecash=0` option.

v0.24.1:

* CLI

A new `bitcoin-cli -generate` command, equivalent to RPC
`generatenewaddress` followed by `generatetoaddress`, can generate
blocks for command line testing purposes. This is a client-side version
of the former `generate` RPC. See the help for details.

* Low-level RPC Changes

- To make RPC `sendtoaddress` more consistent with `sendmany` the
    following error `sendtoaddress` codes were changed from `-4` to
    `-6`:
  - Insufficient funds
  - Transaction has too long of a mempool chain

- Exposed transaction version numbers are now treated as unsigned 32-bit
  integers instead of signed 32-bit integers. This matches their
  treatment in consensus logic. Versions greater than 2 continue to be
  non-standard (matching previous behavior of smaller than 1 or greater
  than 2 being non-standard). Note that this includes the joinpsbt
  command, which combines partially-signed transactions by selecting the
  highest version number.

* Notification changes

`-walletnotify` notifications are now sent for wallet transactions that
are removed from the mempool because they conflict with a new block.
These notifications were sent previously before the v0.21.13 release,
but had been broken since that release.
---
 CMakeLists.txt             | 2 +-
 contrib/aur/lotus/PKGBUILD | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 232bc429c..8eca87420 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ set(CMAKE_USER_MAKE_RULES_OVERRIDE
 )
 
 project(lotus
-	VERSION 1.0.2
+	VERSION 1.0.3
 	DESCRIPTION "Lotus is a full node implementation of the Lotus protocol."
 	HOMEPAGE_URL "https://www.givelotus.org"
 )
diff --git a/contrib/aur/lotus/PKGBUILD b/contrib/aur/lotus/PKGBUILD
index 0ac1db5ca..d8c70e60f 100644
--- a/contrib/aur/lotus/PKGBUILD
+++ b/contrib/aur/lotus/PKGBUILD
@@ -1,7 +1,7 @@
 # Maintainer: Josh Ellithorpe <quest@mac.com>
 
 pkgname=lotusd
-pkgver=1.0.2
+pkgver=1.0.3
 pkgrel=0
 pkgdesc="Lotus with lotusd, lotus-tx, lotus-seeder and lotus-cli"
 arch=('i686' 'x86_64')
-- 
2.39.5 (Apple Git-154)

